department-of-veterans-affairs/vets-api
# ARF 84526 Create mock user data for testing ARP

## Summary

- This work is behind a feature toggle (flipper): NO
- Changes to seed data for local manual testing of the ARP portal.
- Not a bug.
- The solution is to create seed data needed to log in as a veteran representative with the new fields the representative user has.
- Benefits Accredited Rep Facing team. Yes, we own maintenance of this.
- No flipper.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/84526

## Testing done

  - No unit tests needed for seed data and rake tasks.
- Old behavior: no seed data available for local testing.
- Local test steps:
  - Run `db:seed` in vets-api.
  - In `vets-api/app/services/accredited_representative_portal/representative_user_loader.rb:20`, add `puts @current_user` or a binding.pry to inspect the @current_user
  - Start vets-api and vets-website.
  - Navigate to http://localhost:3001/representative/
  - Log in via ID.me using the `vets.gov.user+1` email account.
  - Inspect the vets-api console output for the @current_user. Verify that the `RepresentativeUser` has both `poa_codes` and an `ogc_registration_number`.
- Not behind a flipper.

## Screenshots
<img width="745" alt="Screenshot 2024-07-10 at 12 28 34 PM" src="https://github.com/department-of-veterans-affairs/vets-api/assets/3535749/31ceebcb-9963-48ea-bd0f-60eb28647e7e">


## What areas of the site does it impact?
Local testing of the ARP.

## Acceptance criteria

- [N/A]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [N/A]  Events are being sent to the appropriate logging solution
- [N/A]  Documentation has been updated (link to documentation)
- [N/A]  Feature/bug has a monitor built into Datadog (if applicable)
